File: Python/백준/Bronze/31962. 등교/등교.py
a,b = map(int,input().split())
aa=[]; bb=[]
for i in range(a):
    c,d = map(int,input().split())
    if(c+d<=b):
        aa.append(c)
        bb.append(c+d)
if(len(aa)==0):
    print(-1)
else:
    q = bb.index(max(bb))
    print(aa[q])

File: C#/백준/Bronze/31962. 등교/등교.cs
using System;

namespace q{
    class w{
        static void Main(){
            int[] aa = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            int[] ans = new int[aa[0]];
            for (int i =0; i<aa[0]; i++){
                int[] bb = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
                if(bb[0]+bb[1]<=aa[1]){
                    ans[i]=bb[0];
                }
            }
            if(ans.Sum()==0){
                Console.WriteLine(-1);
            }
            else{
                Console.WriteLine(ans.Max());
            }
        }
    }
}